18_regexp_lab/01.py

```diff
@@ -1,12 +1,21 @@
-"""
-Write a program that reads data
-from property files.
-Each line in the file can either be:
-    An empty line
-    A comment line (Start with #)
-    A property line (of the form key = value)
-
-Write a program that takes a property file name and key
-as command line arguments and prints the requested value
-"""
-
+#Definition file
+
+import re
+import sys
+
+(defFile, key) = sys.argv[1:]
+defDict = {}
+
+#Insert value into the dictionary
+with open(defFile, "r") as fin:
+    for line in fin:
+        word = line.strip(' \t\n\r')
+        if not word.startswith('#'):
+            keyvalue = re.split(r'[=]', line)
+            defDict[keyvalue[0].strip(' \t\n\r')] = keyvalue[1].strip(' \t\n\r')
+
+#Print the value that mact
+if key in defDict.keys():
+    print defDict[key]
+else:
+    print "%s is not in the definition file" % (key)

```

Comment:
Good direction but if the propertise file has an empty line I get
```
Traceback (most recent call last):
  File "01.py", line 15, in <module>
    defDict[keyvalue[0].strip(' \t\n\r')] = keyvalue[1].strip(' \t\n\r')
IndexError: list index out of range
```

Also you could replace ```keyvalue = re.split(r'[=]', line)``` by
``key, value = keyvalue = re.split(r'[=]', line)``` and that would make the code more readable.